### PR TITLE
Add callbacks to dialog

### PR DESCRIPTION
**Pull request by fiffty**
This adds tests for our dialog components, and adds `onEscButtonClick` callback prop to `<DialogFullScreen />`.

`onEscButtonClick` is a callback function that gets called on clicking the escape button. If it returns `false`, it will to prevent dialog from closing.

### CODE REVIEW

File: src/Input/InputFile.tsx [hunk 1]
```diff
@@ -5,7 +5,8 @@ import styled from "styled-components";
 import { useContent, WithContentProps } from "../Content/Content";
 import { AiOutlineClose } from "react-icons/ai";
 import { colors } from "../shared/constants";
-const fileDropIcon = require("../static/file-drop.png");
+// const fileDropIcon = require("../static/file-drop.png");
```
Comment (bcowell) [inline src/Input/InputFile.tsx:8]: Can we remove this comment?

File: src/Dialog/DialogFullScreen.tsx [hunk 2]
```diff
@@ -74,21 +74,35 @@ export const DialogFullScreen = ({
...
   ...rest
 }: DialogProps) => {
   const { lang, theme } = useContent();
+
+  const handleEscButtonClick = useCallback(
+    e => {
+      e.preventDefault();
+
+      if (!onEscButtonClick) return hide(2);
+      if (onEscButtonClick(e)) return hide(2);
```
Comment (bcowell) [inline src/Dialog/DialogFullScreen.tsx:89]: hide(e)

File: stories/components/Dialog/Dialog.md [hunk 3]
```diff
@@ -47,11 +54,15 @@ A light-grey coloured background that takes up the full screen.
...
+
+| name               | type         | default | description                                                                                                     |
+| ------------------ | ------------ | ------- | --------------------------------------------------------------------------------------------------------------- |
+| label              | string       |         | Aria-label                                                                                                      |
+| children           | Component    |         | Placeholder for input                                                                                           |
+| hideOnEsc          | boolean      | true    | Closes dialog on esc press if true                                                                              |
+| hideOnClickOutside | boolean      | true    | Closes dialog on clicking outside of dialog if true                                                             |
+| onEscButtonClick   | e => boolean |         | Callback function that gets called on clicking the escape button. Return `false` to prevent dialog from closing |
```
Comment (bcowell) [inline stories/components/Dialog/Dialog.md:65]: nit: Lines don't need the trailing spaces and very last | to render as a table in markdown. Would make it look better reading from an editor

Comment (fiffty) [reply stories/components/Dialog/Dialog.md:65]: I think this is prettier formatting it automatically

**Pull request merged**